### PR TITLE
Sort getEvent

### LINKED ISSUE #57: Correctly sort matches in event

### PR DESCRIPTION
**Pull request by GreenBeard**
Closes #57

### CODE REVIEW

File: src/api.tsx [hunk 1]
```diff
@@ -36,7 +36,32 @@ const get = <T extends {}>(url: string) => async (
...
+const getEvent = (eventKey: string) => async (
+  cb: (err: Error | null, data: FRCEvent | null) => any
+) => {
+  get<FRCEvent>(`events/${eventKey}`)((err: Error, data: FRCEvent) => {
+    if (err) {
+      cb(err, data)
+    } else {
+      if (data != null && data.matches != null) {
+        data.matches = data.matches
+          .map((value: Match) => {
```
Comment (calebeby) [inline src/api.tsx]: Types are inferred

File: src/api.tsx [hunk 2]
```diff
@@ -36,7 +36,32 @@ const get = <T extends {}>(url: string) => async (
...
 
-const getEvent = (eventKey: string) => get<FRCEvent>(`events/${eventKey}`)
+const getEvent = (eventKey: string) => async (
+  cb: (err: Error | null, data: FRCEvent | null) => any
+) => {
+  get<FRCEvent>(`events/${eventKey}`)((err: Error, data: FRCEvent) => {
+    if (err) {
+      cb(err, data)
+    } else {
+      if (data != null && data.matches != null) {
```
Comment (calebeby) [inline src/api.tsx]: Triple equals

Comment (GreenBeard) [reply src/api.tsx]: What if it is undefined? Or should I explicitly check for both with the triple equals?

Comment (calebeby) [reply src/api.tsx]: When will it be undefined?

**Comment by calebeby** [comment]
@GreenBeard is it going to be undefined or null?

File: src/api.tsx [hunk 3]
```diff
@@ -36,7 +36,32 @@ const get = <T extends {}>(url: string) => async (
...
+      if (data !== undefined && data.matches !== undefined) {
+        data.matches = data.matches
+          .map(value => {
+            if (!(value.time instanceof Date)) {
+              value.time = new Date(value.actualTime || value.predictedTime)
+            }
+            return value
+          })
+          .sort((a, b) => {
+            return (
```
Comment (calebeby) [inline src/api.tsx]: Can you make this inline since it is only one "line"?

**Pull request merged**